sound/soc/sof/pm.c

```diff
@@ -411,3 +411,21 @@ int snd_sof_prepare(struct device *dev)
 	return 0;
 }
 EXPORT_SYMBOL(snd_sof_prepare);
+
+void snd_sof_pm_runtime_enable(struct snd_sof_dev *sdev)
+{
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(sdev->dev, SND_SOF_SUSPEND_DELAY);
+	pm_runtime_use_autosuspend(sdev->dev);
+
+	/*
+	 * runtime pm for pci device is "forbidden" by default.
+	 * so call pm_runtime_allow() to enable it.
+	 */
+	pm_runtime_allow(sdev->dev);
+
+	/* follow recommendation in pci-driver.c to decrement usage counter */
+	if (sdev->pci)
+		pm_runtime_put_noidle(sdev->dev);
```

Comment:
@lgirdwood and here. this should also be pm_runtime_put_nodle(&sdev->pci->dev)

Reply:
@lgirdwood yes that was my point, you need to be careful on which  device you interact with. The common routine only handled sdev->dev.

Reply:
@lgirdwood sure, I will